src/scripts/main.js

```diff
@@ -1,3 +1,385 @@
+/* eslint-disable max-len */
 'use strict';
 
-// write code here
+const NAME = 'Name';
+const POSITION = 'Position';
+const OFFICE = 'Office';
+const AGE = 'Age';
+const SALARY = 'Salary';
+
+const DONE = 'done';
+const NOT_DONE = 'not-done';
+
+const body = document.getElementsByTagName('body')[0];
+const tbody = document.getElementsByTagName('tbody')[0];
+const headers = document.querySelectorAll('th');
+const topHeaders = Array.from(headers).slice(0, 5);
+let table = tbody.querySelectorAll('tr');
+const staff = [];
+
+let isClickedName = false;
+let isClickedPosition = false;
+let isClickedOffice = false;
+let isClickedAge = false;
+let isClickedSalary = false;
+
+table.forEach((row) => {
+  const memberStaff = {};
+
+  memberStaff.name = row.querySelectorAll('td')[0].textContent;
+  memberStaff.position = row.querySelectorAll('td')[1].textContent;
+  memberStaff.office = row.querySelectorAll('td')[2].textContent;
+  memberStaff.age = row.querySelectorAll('td')[3].textContent;
+  memberStaff.salary = row.querySelectorAll('td')[4].textContent;
+
+  staff.push(memberStaff);
+});
+
+topHeaders.forEach((header) => {
+  header.addEventListener('click', (e) => {
+    switch (e.target.textContent) {
+      case NAME:
+        const acsSortedName = staff.sort((a, b) =>
+          a.name.localeCompare(b.name),
+        );
+
+        const sortedByName = isClickedName
+          ? acsSortedName.reverse()
+          : acsSortedName;
+
+        table.forEach((row, index) => {
+          row.querySelectorAll('td')[0].textContent = sortedByName[index].name;
+
+          row.querySelectorAll('td')[1].textContent =
+            sortedByName[index].position;
+
+          row.querySelectorAll('td')[2].textContent =
+            sortedByName[index].office;
+
+          row.querySelectorAll('td')[3].textContent = sortedByName[index].age;
+
+          row.querySelectorAll('td')[4].textContent =
+            sortedByName[index].salary;
+        });
+
+        isClickedName = !isClickedName;
+
+        break;
+
+      case POSITION:
+        const acsSortedPosition = staff.sort((a, b) =>
+          a.position.localeCompare(b.position),
+        );
+
+        const sortedByPosition = isClickedPosition
+          ? acsSortedPosition.reverse()
+          : acsSortedPosition;
+
+        table.forEach((row, index) => {
+          row.querySelectorAll('td')[0].textContent =
+            sortedByPosition[index].name;
+
+          row.querySelectorAll('td')[1].textContent =
+            sortedByPosition[index].position;
+
+          row.querySelectorAll('td')[2].textContent =
+            sortedByPosition[index].office;
+
+          row.querySelectorAll('td')[3].textContent =
+            sortedByPosition[index].age;
+
+          row.querySelectorAll('td')[4].textContent =
+            sortedByPosition[index].salary;
+        });
+
+        isClickedPosition = !isClickedPosition;
+
+        break;
+
+      case OFFICE:
```

Comment:
And here 
<img width="705" alt="image" src="https://github.com/mate-academy/js_employees_table_DOM/assets/64750958/a6af8781-0033-4cc8-a494-37105d096dde">